internal/services/computefleet/compute_fleet_resource.go

```diff
@@ -0,0 +1,1006 @@
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+
+				if model.Identity != nil {
+					v, err := identity.FlattenSystemAndUserAssignedMapToModel(pointer.To(identity.SystemAndUserAssignedMap(*model.Identity)))
+					if err != nil {
+						return err
+					}
+					state.Identity = pointer.From(v)
+				}
+
+				state.Plan = flattenPlanModel(model.Plan)
+
+				if props := model.Properties; props != nil {
+					state.AdditionalCapabilities = flattenAdditionalCapabilities(props.ComputeProfile.AdditionalVirtualMachineCapabilities)
+
+					baseVirtualMachineProfileValue, err := flattenVirtualMachineProfileModel(&props.ComputeProfile.BaseVirtualMachineProfile, metadata)
+					if err != nil {
+						return err
+					}
+					state.VirtualMachineProfile = baseVirtualMachineProfileValue
+
+					state.ComputeApiVersion = pointer.From(props.ComputeProfile.ComputeApiVersion)
+					state.PlatformFaultDomainCount = pointer.From(props.ComputeProfile.PlatformFaultDomainCount)
+					state.RegularPriorityProfile = flattenRegularPriorityProfileModel(props.RegularPriorityProfile)
+					state.SpotPriorityProfile = flattenSpotPriorityProfileModel(props.SpotPriorityProfile)
+					state.UniqueId = pointer.From(props.UniqueId)
+					state.VMSizesProfile = flattenVMSizeProfileModel(&props.VMSizesProfile)
+				}
+				state.Tags = pointer.From(model.Tags)
+				state.Zones = pointer.From(model.Zones)
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r ComputeFleetResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ComputeFleet.ComputeFleetClient
+
+			id, err := fleets.ParseFleetID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ComputeFleetResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var state ComputeFleetResourceModel
+			if err := metadata.DecodeDiff(&state); err != nil {
+				return fmt.Errorf("DecodeDiff: %+v", err)
+			}
+
+			if v := state.Identity; len(v) > 0 {
+				// whilst the Swagger defines multiple at this time only UAI is supported
+				if v[0].Type != identity.TypeUserAssigned {
+					return fmt.Errorf("the type `%s` of `identity` is not supported currently", v[0].Type)
+				}
+			}
+
+			if len(state.SpotPriorityProfile) > 0 && len(state.RegularPriorityProfile) > 0 {
+				if state.SpotPriorityProfile[0].Capacity+state.RegularPriorityProfile[0].Capacity > 10000 {
+					return fmt.Errorf("the sum of `spot_priority_profile.0.capacity` and `regular_priority_profile.0.capacity` must be between `0` and `10000`, inclusive")
```

Comment:
When not using a format string for an error message, you can simplify and use `errors.New()` (make sure to import `errors`)
```suggestion
					return errors.New("the sum of `spot_priority_profile.0.capacity` and `regular_priority_profile.0.capacity` must be between `0` and `10000`, inclusive")
```